src/metrics/whole_scene_coverage.py

```diff
@@ -0,0 +1,34 @@
+import open3d as o3d
+import psutil
+
+from src.core import Database
+from src.metrics.reduction_metric import ReductionMetric
+from tqdm import tqdm
+
+MEMORY_LIMIT = 70
```

Comment:
I am not sure about the selected memory threshold. However, it seems that voxel_down_sample requires a lot of memory to work, so I want to leave some memory for this